File: src/TheWorld/wwwroot/lib/_app/tripsController.js
!function(){"use strict";function r(r){var i=this;i.trips=[],i.newTrip={},i.errorMessage="",i.isBusy=!0,r.get("/api/trips").then(function(r){angular.copy(r.data,i.trips)},function(){i.errorMessage="Failed to load data: "+error}).finally(function(){i.isBusy=!1}),i.addTrip=function(){i.isBusy=!0,i.errorMessage="",r.post("/api/trips",i.newTrip).then(function(r){i.trips.push(r.data),i.newTrip={}},function(){i.errorMessage="Failed to sve new Trip"}).finally(function(){i.isBusy=!1})}}angular.module("app-trips").controller("tripsController",r)}();

File: src/TheWorld/Controllers/Api/TripsController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.Extensions.Logging;
using TheWorld.Models;
using TheWorld.ViewModels;

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    public class TripsController : Controller
    {
        private readonly IWorldRepository _repository;
        private ILogger<TripsController> _logger;


        public TripsController(IWorldRepository repository,ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }
        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var results = _repository.GetAllTrips();
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                // TODO Logging
                _logger.LogError($"Failed to get All Trips: {ex}");
                return BadRequest(ex);
            }
        }

        [HttpPost("")]
        public IActionResult Post([FromBody]TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                // Save to the Database
                var newTrip = Mapper.Map<Trip>(theTrip);

                return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
            }
            return BadRequest(ModelState);
        }
    }
}


